pkg/dogstatsd/enrich.go

```diff
@@ -0,0 +1,199 @@
+package dogstatsd
+
+import (
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/tagger"
+	"github.com/DataDog/datadog-agent/pkg/tagger/collectors"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/kubelet"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type tagRetriever func(entity string, cardinality collectors.TagCardinality) ([]string, error)
+
+var (
+	hostTagPrefix     = "host:"
+	entityIDTagPrefix = "dd.internal.entity_id:"
+
+	getTags tagRetriever = tagger.Tag
+)
+
+func parseMetricMessage(message []byte, namespace string, namespaceBlacklist []string, defaultHostname string) (*metrics.MetricSample, error) {
+	sample, err := parseMetricSample(message)
+	if err != nil {
+		return nil, err
+	}
+	return enrichMetricSample(sample, namespace, namespaceBlacklist, defaultHostname), nil
```

Comment:
Is that the location where the caching and mapping happen? just before `enrichMetricSample` (similar for other type of samples)

Reply:
I'd say it depends on your use case. The caching I plan to work on targets parsing. I think you can plug yours between the two or directly in enrichment.